sdk/identity/identity/src/client/msalClient.ts

```diff
@@ -0,0 +1,284 @@
+      cachePath,
+      "serviceName",
+      "accountName"
+    );
+
+    return {
+      cachePlugin: new msalExt.PersistenceCachePlugin(libSecretPersistence)
+    };
+  }
+
+  // fall back to using plain text file. Not recommended for storing secrets.
+  const filePersistence = await msalExt.FilePersistence.create(cachePath);
+
+  return {
+    cachePlugin: new msalExt.PersistenceCachePlugin(filePersistence)
+  };
+}
+
+/**
+ * The record to use to find the cached tokens in the cache
+ */
+export interface AuthenticationRecord {
+  /**
+   * The associated authority, if used
+   */
+  authority?: string;
+
+  /**
+   * The home account Id
+   */
+  homeAccountId: string;
+
+  /**
+   * The login environment, eg "login.windows.net"
+   */
+  environment: string;
+
+  /**
+   * The associated tenant ID
+   */
+  tenantId: string;
+
+  /**
+   * The username of the logged in account
+   */
+  username: string;
+}
+
+export class AuthenticationRequired extends CredentialUnavailable {}
+
+export class MsalClient {
+  private persistenceEnabled: boolean;
+  private authenticationRecord: AuthenticationRecord | undefined;
+  private identityClient: IdentityClient;
+  private pca: PublicClientApplication | undefined;
+  private cca: ConfidentialClientApplication | undefined;
+  private cachePath?: string;
+  private msalConfig: NodeAuthOptions;
+
+  constructor(
+    msalConfig: NodeAuthOptions,
+    persistenceEnabled: boolean,
+    authenticationRecord?: AuthenticationRecord,
+    cachePath?: string,
+    options?: TokenCredentialOptions
+  ) {
+    this.identityClient = new IdentityClient(options);
+    this.msalConfig = msalConfig;
+    this.cachePath = cachePath ? cachePath : "cache.bin";
+    this.persistenceEnabled = persistenceEnabled;
+    this.authenticationRecord = authenticationRecord;
+  }
+
+  async prepareClientApplications() {
+    // If we've already initialized the public client application, return
+    if (this.pca) {
+      return;
+    }
+
+    // If we need to load the plugin that handles persistence, go ahead and load it
+    let plugin = undefined;
+    if (this.persistenceEnabled && this.authenticationRecord) {
+      plugin = await createPersistence(this.cachePath);
+    }
+
+    // Construct the public client application, since it hasn't been initialized, yet
+    const clientConfig: Configuration = {
+      auth: this.msalConfig,
+      cache: plugin,
+      system: { networkClient: this.identityClient }
+    };
+
+    this.pca = new PublicClientApplication(clientConfig);
+  }
+
+  async acquireTokenFromCache(): Promise<AccessToken | null> {
+    await this.prepareClientApplications();
+
+    if (!this.persistenceEnabled || !this.authenticationRecord) {
+      throw new AuthenticationRequired();
+    }
+
+    const silentRequest = {
+      account: this.authenticationRecord!,
+      scopes: ["https://vault.azure.net/user_impersonation", "https://vault.azure.net/.default"]
```

Comment:
Why is this defaulting to acquiring a token for key vault? Shouldn't the scope be passed into `acquireTokenFromCache`?